classes/Parameters/LocalChannelConfigurationValidator.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace PrestaShop\Module\AutoUpgrade\Parameters;
+
+use Exception;
+use PrestaShop\Module\AutoUpgrade\Services\PrestashopVersionService;
+use PrestaShop\Module\AutoUpgrade\UpgradeTools\Translator;
+use PrestaShop\Module\AutoUpgrade\ZipAction;
+
+class LocalChannelConfigurationValidator
+{
+    /**
+     * @var Translator
+     */
+    private $translator;
+
+    /**
+     * @var ZipAction
+     */
+    private $zipAction;
+
+    /**
+     * @var string
+     */
+    private $downloadPath;
+
+    public function __construct(Translator $translator, ZipAction $zipAction, string $downloadPath)
+    {
+        $this->translator = $translator;
+        $this->zipAction = $zipAction;
+        $this->downloadPath = $downloadPath;
+    }
+
+    /**
+     * @param array<string, mixed> $array
+     *
+     * @return array<string, string>
+     */
+    public function validate(array $array = []): array
+    {
+        $file = $array['archive_zip'];
+        $fullFilePath = $this->downloadPath . DIRECTORY_SEPARATOR . $file;
+        if (!file_exists($fullFilePath)) {
+            return ['archive_zip' => $this->translator->trans('File %s does not exist. Unable to select that channel.', [$file])];
+        }
+
+        try {
+            $targetVersion = (new PrestashopVersionService($this->zipAction))->extractPrestashopVersionFromZip($fullFilePath);
+        } catch (Exception $exception) {
+            return ['archive_zip' => $this->translator->trans('We couldn\'t find a PrestaShop version in the .zip file that was uploaded in your local archive. Please try again.')];
+        }
+
+        $xmlFile = $array['archive_xml'];
+        $fullXmlPath = $this->downloadPath . DIRECTORY_SEPARATOR . $xmlFile;
+        if (!empty($xmlFile) && !file_exists($fullXmlPath)) {
+            return ['archive_xml' => $this->translator->trans('File %s does not exist. Unable to select that channel.', [$xmlFile])];
+        }
+
+        try {
+            $xmlVersion = (new PrestashopVersionService($this->zipAction))->extractPrestashopVersionFromXml($fullXmlPath);
+        } catch (Exception $exception) {
+            return ['archive_xml' => $this->translator->trans('We couldn\'t find a PrestaShop version in the XML file that was uploaded in your local archive. Please try again')];
+        }
+
+        if ($xmlVersion !== $targetVersion) {
+            return ['global' => $this->translator->trans('The PrestaShop version in your archive doesn’t match the one in XML file. Please fix this issue and try again.')];
```

Comment:
Why “global”? Why not version or “prestashop_version”?

Reply:
it's more about the frontend part we don't know how to name it because he is not attached to an unique input.

Reply:
this error can also be present on the CLI version, and I find that the "global" key is not very meaningful for this error

Reply:
yes but in CLI you never used the key.

Reply:
For the moment yes, but the key still has no meaning in relation to the error it throws

Reply:
and what do you think about `form` ?

Reply:
I don't understand x) The errors for zip have "archive_zip", those for xml have "archive_xml", so why the error for version matching can't be "version" or "file_version"?

Reply:
why `global` and `form` ?

Reply:
cause we don't have any field named 'version' or 'file_version'.

Reply:
A validator shouldn’t contain responses for specific front-end requirements, right? It's fine to return keys or error codes, but front-end technical needs shouldn’t alter the validator's behavior